tests/test_main.py

```diff
@@ -258,6 +305,50 @@ async def test_get_generator_data(
     assert response == snapshot
 
 
+@pytest.mark.parametrize(
+    "fixture_file,method,generator_id,expected_url",
+    (
+        (
+            "alerts_rdc2v4.json",
+            "get_alerts",
+            12345,
+            f"{API_BASE}/kem/api/v3/devices/12345/alerts",
+        ),
+        (
+            "events_rdc2v4.json",
+            "get_events",
+            12345,
+            f"{API_BASE}/kem/api/v3/devices/12345/events",
+        ),
+        (
+            "maintenance_notes.json",
+            "get_maintenance_notes",
+            12345,
+            f"{API_BASE}/kem/api/v3/devices/12345/maintenance_notes",
+        ),
+    ),
+)
+async def test_generator_endpoints(
+    fixture_file: str, method: str, generator_id: int, expected_url: str
+) -> None:
+    # Create a mock session
+    mock_session = Mock()
+    kem = await get_kem(mock_session)
+    kem.set_timeout(5)
+    mock_response = AsyncMock()
+    mock_response.status = 200
+    mock_response.json.return_value = load_fixture_file(fixture_file)
+    mock_session.get.return_value = mock_response
+
+    result = await getattr(kem, method)(generator_id)
+
+    # Assert that the session.get method was called with the correct URL and data
+    mock_session.get.assert_called_once()
+    assert str(mock_session.get.call_args[0][0]) == expected_url
+
+    assert result == mock_response.json.return_value
+
+
```

Comment:
This test will test the exception on a bad return type; which should improve the codecov and make that test pass.

```suggestion
@pytest.mark.parametrize(
    "method,generator_id",
    (
        (
            "get_alerts",
            12345,
        ),
        (
            "get_events",
            12345,
        ),
        (
            "get_maintenance_notes",
            12345,
        ),
    ),
)
async def test_generator_endpoints_bad_type(
    method: str, generator_id: int) -> None:
    # Create a mock session
    mock_session = Mock()
    kem = await get_kem(mock_session)
    kem.set_timeout(5)
    mock_response = AsyncMock()
    mock_response.status = 200
    mock_response.json.return_value = "invalid response"
    mock_session.get.return_value = mock_response

    with pytest.raises(TypeError):
        _ = await getattr(kem, method)(generator_id)
```